### PR TITLE
RavenDB-15108

### PR DESCRIPTION
**Pull request by aviv86**
https://issues.hibernatingrhinos.com/issue/RavenDB-15108

### CODE REVIEW

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs [hunk 1]
```diff
@@ -142,13 +148,20 @@ public void UpdateStats(DocumentsOperationContext context, TimeSeriesSliceHolder
...
             if (table.ReadByKey(slicer.StatsKey, out var tvr))
             {
                 previousCount = DocumentsStorage.TableValueToLong((int)StatsColumns.Count, ref tvr);
                 start = new DateTime(Bits.SwapBytes(DocumentsStorage.TableValueToLong((int)StatsColumns.Start, ref tvr)));
                 end = DocumentsStorage.TableValueToDateTime((int)StatsColumns.End, ref tvr);
+                nameScope = DocumentsStorage.TableValueToSlice(context, (int)StatsColumns.Name, ref tvr, out nameSlice);
+            }
+            else
+            {
+                nameScope = Slice.From(context.Allocator, slicer.Name, out nameSlice);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs]: Let the slicer hold also the original nameSlice

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs [hunk 2]
```diff
@@ -268,6 +283,39 @@ private static DateTime GetLastLiveTimestamp(DocumentsOperationContext context,
...
+            using (DocumentIdWorker.GetSliceFromId(context, docId, out var documentKeyPrefix, SpecialChars.RecordSeparator))
+            {
+                foreach (var result in table.SeekByPrimaryKeyPrefix(documentKeyPrefix, Slices.Empty, 0))
+                {
+                    yield return DocumentsStorage.TableValueToChangeVector(context, (int)StatsColumns.Name, ref result.Value.Reader);
+                }
+            }
+        }
+
+        public LazyStringValue GetTimeSeriesNameOriginalCasing(DocumentsOperationContext context, Slice storageKey)
```
Comment (karmeli87) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs]: do we need all of the different `GetOriginalName` if we have this method?

File: test/SlowTests/Client/TimeSeries/TimeSeriesAndDocumentConflicts.cs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+using Raven.Client;
+using Raven.Client.Documents.Operations.TimeSeries;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents;
+using Raven.Tests.Core.Utils.Entities;
+using Sparrow;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Client.TimeSeries
```
Comment (karmeli87) [inline test/SlowTests/Client/TimeSeries/TimeSeriesAndDocumentConflicts.cs:14]: What if we have a conflict on a time-series casing?
e.g. "Heartrate" vs "HeartRate"

**Review by karmeli87 — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs (hunk 1 above)
Comment (aviv86) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs]: done

File: test/SlowTests/Client/TimeSeries/TimeSeriesAndDocumentConflicts.cs (hunk 3 above)
Comment (aviv86) [reply test/SlowTests/Client/TimeSeries/TimeSeriesAndDocumentConflicts.cs:14]: handled

**Review by karmeli87 — APPROVED** [review]
LGTM

**Review by ppekrol — APPROVED** [review]

**Pull request merged**